sdk/appconfiguration/azure-data-appconfiguration/src/test/java/com/azure/data/appconfiguration/ConfigurationClientTestBase.java

```diff
@@ -412,9 +411,32 @@ static void assertConfigurationEquals(ConfigurationSetting expected, Response<Co
     static void assertConfigurationEquals(ConfigurationSetting expected, ConfigurationSetting actual) {
         if (expected != null && actual != null) {
             actual = cleanResponse(expected, actual);
+        } else if (expected == actual) {
+            return;
+        } else if (expected == null || actual == null) {
+            assertTrue(false);
         }
 
-        assertEquals(expected, actual);
+        assertEquals(expected.getKey(), actual.getKey());
+        assertEquals(expected.getLabel(), actual.getLabel());
+        assertEquals(expected.getValue(), actual.getValue());
+        assertEquals(expected.getETag(), actual.getETag());
+        assertEquals(expected.getLastModified(), actual.getLastModified());
+        assertEquals(expected.getContentType(), actual.getContentType());
+
+        final Map<String, String> expectedTags = expected.getTags();
+        final Map<String, String> actualTags = actual.getTags();
+
+        boolean expectedIsNullOrEmpty = ImplUtils.isNullOrEmpty(expectedTags);
+        boolean actualIsNullOrEmpty = ImplUtils.isNullOrEmpty(actualTags);
+
+        if (expectedIsNullOrEmpty) {
```

Comment:
can you do the following instead of this? It's hard to follow... especially with that assertTRue(true) down there.

```java
assertEquals(expectedIsNullOrEmpty, actualIsNullOrEmpty);
assertEquals(expectedTags, actualTags);
```